test-case/check-runtime-pm-boundary.sh

```diff
@@ -0,0 +1,165 @@
+    ;;
+    "capture")
+        cmd=arecord
+        dummy_file=/dev/null
+    ;;
+    *)
+        dloge "Invalid test mode: $test_mode (allow value : playback, capture)"
+        exit 1
+    ;;
+esac
+
+runtime_status="/sys/bus/pci/devices/0000:$(lspci |awk '/[Aa]udio/ {print $1;}')/power/runtime_status"
+
+[[ ! -f $runtime_status ]] && dloge "no runtime_status entry: $runtime_status" && exit 1
+
+dlogc "Runtime status check: cat $runtime_status"
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+func_pipeline_export $tplg "type:$test_mode"
+func_lib_setup_kernel_last_line
+
+arr_sleep=(3 5 7 10 15)
+for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+do
+    channel=$(func_pipeline_parse_value $idx channel)
+    rate=$(func_pipeline_parse_value $idx rate)
+    fmt=$(func_pipeline_parse_value $idx fmt)
+    dev=$(func_pipeline_parse_value $idx dev)
+    pcm=$(func_pipeline_parse_value $idx pcm)
+
+    for pm_sleep in ${arr_sleep[@]}
+    do
+        for i in $(seq 1 $loop_count)
+        do
+            dlogi "Iteration [$idx] [$i of $loop_count] for $pcm, [sleep $pm_sleep]"
+            # playback device - check status
+            dlogc "$cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q"
+            $cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q &
+            pid=$!
+
+            # TODO: delay 2.5s is workaround for the SSH aplay delay issue.
+            sleep 2.5
+
+            kill -0 $pid
+            [[ $? -ne 0 ]] && dloge "$cmd process for pcm $pcm is not alive" && exit 1
+
+            [[ -d /proc/$pid ]] && result=`cat $runtime_status`
+
+            dlogi "While $cmd running, runtime status: $result"
+            if [[ $result == active ]]; then
+                # stop playback device - check status again
+                dlogc "kill process: kill -9 $pid"
+                kill -9 $pid && wait $pid 2>/dev/null
+                dlogi "$cmd end"
+
+                dlogc "sleep $pm_sleep"
+                sleep $pm_sleep
+
+                result=`cat $runtime_status`
+                dlogi "After $cmd killed and sleep $pm_sleep, runtime status: $result"
+                if [ $pm_sleep -gt $state_delay ] && [ $result != suspended ]; then
+                    # check runtime pm status ONLY when pm_sleep is greater than expected state transition time
+                    dloge "$cmd process for pcm $pcm runtime status is not suspended as expected"
+                    exit 1
+                fi
+
+                dlogi "check $cmd can start without any problem after sleep $pm_sleep"
+                dlogc "$cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q"
+                $cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q &
+                pid=$!
+
+                sleep 1
+
+                kill -0 $pid
+                [[ $? -ne 0 ]] && dloge "$cmd process for pcm $pcm is not alive" && exit 1
+
+                kill -9 $pid && wait $pid 2>/dev/null
+                dlogi "$cmd end"
```

Comment:
I don't think this operation is different with case: [check-runtime-pm-status](https://github.com/thesofproject/sof-test/blob/master/test-case/check-runtime-pm-status.sh)  

you can see when you run this logic in the loop, so next round will run the cmd again  
So I will suggest you add the **loop** option for the original case instead of this change  
if `$i` try to run and trigger the error at here, I think original case will trigger at `$i + 1` at begin  
when add **loop** option